### PR TITLE
patches for qt-creator trunk (and 3.4)

### PR DESCRIPTION
**Pull request by jhunold**
Adapt to change in QtCreator.
- changes in MimeData handling
- Some more explicit QString vs. Utils::Filename conversions

### CODE REVIEW

**Comment by jhunold** [comment]
Shall we discuss integration into QtCreator on GitHub, on the QtCreator-ML or via direct email?

**Comment by mloskot** [comment]
@jhunold let's discuss the preparation part here on GitHub, hence I created #13 for that purpose. Once the plug-in code is submitted, then we close #13 and follow on Gerrit and/or QtCreator-ML.

File: boostbuildprojectmanager/bbprojectnode.cpp [hunk 1]
```diff
@@ -228,9 +228,9 @@ ProjectNode::createFolderByName(QStringList const& components, int const end)
         return this;
 
     using ProjectExplorer::FolderNode;
-    QString const baseDir = QFileInfo(path()).path();
+    QString const baseDir = QFileInfo(path().toString()).path();
```
Comment (orgads) [inline boostbuildprojectmanager/bbprojectnode.cpp]: path().parentDir().toString()

File: boostbuildprojectmanager/bbprojectnode.cpp [hunk 2]
```diff
@@ -253,10 +253,10 @@ ProjectNode::findFolderByName(QStringList const& components, int const end) cons
         return 0;
 
     QString const folderName = appendPathComponents(components, end);
-    QString const baseDir = QFileInfo(path()).path();
+    QString const baseDir = QFileInfo(path().toString()).path();
```
Comment (orgads) [inline boostbuildprojectmanager/bbprojectnode.cpp]: ditto

**Comment by jhunold** [comment]
@orgads Thanks for the hints, I'll try and clean up the patch over the weekend. My first goal was to get it compile.

**Comment by mloskot** [comment]
I will also apply @orgads' suggestions, manually.

**Pull request merged**